hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/GrpcBackendResolver.java

```diff
@@ -25,19 +25,20 @@
     Protocol protocol = EnrichedSpanUtils.getProtocol(event);
 
     if (protocol == Protocol.PROTOCOL_GRPC) {
-      String backendURI = SpanAttributeUtils.getStringAttribute(event,
-          Constants.getRawSpanConstant(Grpc.GRPC_HOST_PORT));
+      Optional<String> backendURI = RpcSemanticConventionUtils
+          .getGrpcURI(event);
 
-      if (StringUtils.isEmpty(backendURI)) {
+      if (backendURI.isEmpty() || StringUtils.isEmpty(backendURI.get())) {
```

Comment:
This is confusing, do we need these two checks? how come?

Reply:
`backendURI` is an Optional var.